### ISSUE TITLE
Surprising behavior on `instantiate`

### ISSUE THREAD

**Issue by 00vareladavid** [JuliaLang/Pkg.jl#1678]
Perhaps I'm missing something, but the following behavior surprised me. Shouldn't the artifacts have downloaded on `instantiate` and not on some later operation?

```
(qpoiwer) pkg> add FreeType2_jll
   Updating registry at `~/.julia/registries/General`
   Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Resolving package versions...
   Updating `/tmp/qpoiwer/Project.toml`
  [d7e528f0] + FreeType2_jll v2.10.1+1
   Updating `/tmp/qpoiwer/Manifest.toml`
  [6e34b625] + Bzip2_jll v1.0.6+1
  [d7e528f0] + FreeType2_jll v2.10.1+1
  [83775a58] + Zlib_jll v1.2.11+8
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 

julia> push!(empty!(DEPOT_PATH), "/tmp/depot")
1-element Array{String,1}:
 "/tmp/depot"

(qpoiwer) pkg> instantiate
    Cloning default registries into `/tmp/depot`
######################################################################## 100.0%
      Added registry `General` to `/tmp/depot/registries/General`
  Installed FreeType2_jll ─ v2.10.1+1
  Installed Zlib_jll ────── v1.2.11+8
  Installed Bzip2_jll ───── v1.0.6+1

(qpoiwer) pkg> dev Example
Path `/home/david/pkg/Example` exists and looks like the correct package. Using existing path.
  Resolving package versions...
Downloading artifact: Zlib
Downloading artifact: Bzip2
Downloading artifact: FreeType2
   Updating `/tmp/qpoiwer/Project.toml`
  [7876af07] + Example v0.5.4 [`~/pkg/Example`]
   Updating `/tmp/qpoiwer/Manifest.toml`
  [7876af07] + Example v0.5.4 [`~/pkg/Example`]
```

**Comment by fredrikekre** [comment]
Looks like https://github.com/JuliaLang/Pkg.jl/issues/1338

**Comment by 00vareladavid** [comment]
It does. But the fix for that should be in 1.4 rc1, correct?

**Comment by KristofferC** [comment]
Fixed by https://github.com/JuliaLang/Pkg.jl/pull/1777.
